integration_test/test_xgboost_run.py

```diff
@@ -145,31 +149,93 @@ def _verify_metric_logging(self, run):
+      else:
+        assert len(run.datasets) == 1
+        assert run.datasets[0] == DEFAULT_EVALS_NAME
+
+    assert XGB_INTEGRATION_KEYWORD in run.dev_metadata
+    assert run.dev_metadata[XGB_INTEGRATION_KEYWORD]
+
+  def _verify_checkpoint_logging(self, run):
+    if not self.run_params['verbose_evals']:
+      assert run.checkpoint_count == (
+        self.run_params['num_boost_round'] // DEFAULT_CHECKPOINT_PERIOD + 1 * (
+          self.run_params['num_boost_round'] % DEFAULT_CHECKPOINT_PERIOD > 0
+        )
+      )
+    elif self.run_params['verbose_evals'] is True:
+      assert run.checkpoint_count == self.run_params['num_boost_round']
+    else:
+      assert run.checkpoint_count == (
+        self.run_params['num_boost_round'] // self.run_params['verbose_evals'] + 1 * (
+          self.run_params['num_boost_round'] % self.run_params['verbose_evals'] > 0
+        )
+      )
+
   @pytest.mark.parametrize("task", ['binary', 'multiclass', 'regression'])
   def test_run(self, task):
     self.is_classification = True if task in ('binary', 'multiclass') else False
     self.run_params = _form_random_run_params(task)
     ctx = sigopt.xgboost.run(**self.run_params)
     run = sigopt.get_run(ctx.run.id)
-    if 'name' in self.run_params['run_options']:
-      assert run.name == self.run_params['run_options']['name']
+
     self._verify_metadata_logging(run)
     self._verify_parameter_logging(run)
     self._verify_metric_logging(run)
+    self._verify_feature_importances_logging(run, ctx.model)
+    self._verify_miscs_data_logging(run)
+
+  def test_run_options_no_logging(self):
+    self.run_params = _form_random_run_params(task='binary')
+    self.run_params['run_options'].update({
+      'log_checkpoints': False,
+      'log_feature_importances': False,
+      'log_metrics': False,
+      'log_params': False,
+    })
+    ctx = sigopt.xgboost.run(**self.run_params)
+    run = sigopt.get_run(ctx.run.id)
 
-    real_scores = sorted(ctx.model.get_score(importance_type='weight').items(), key=lambda x: (x[1], x[0]), reverse=True)
-    if real_scores:
-      feature_importances = run.sys_metadata['feature_importances']
-      saved_scores = sorted(feature_importances['scores'].items(), key=lambda x: (x[1], x[0]), reverse=True)
-      assert feature_importances['type'] == 'weight'
-      assert saved_scores and len(saved_scores) <= len(real_scores)
-      real_scores = real_scores[:len(saved_scores)]
-      assert [k for k, v in real_scores] == [k for k, v in saved_scores]
-      assert numpy.allclose(
-        numpy.array([v for k, v in real_scores]),
-        numpy.array([v for k, v in saved_scores]),
+    assert run.checkpoint_count == 0
+    assert 'feature_importances' not in run.sys_metadata
+    assert set(run.values.keys()) == set(
+      '-'.join((data_name, metric_name)) for data_name, metric_name in itertools.product(
+        self.run_params['evals'], self.run_params['params']['eval_metric']
```

Comment:
Test isn't passing atm, should be: 

`  assert set(run.values.keys()) >= set(
      ['-'.join((data_name[1], metric_name)) for data_name, metric_name in itertools.product(
        self.run_params['evals'], self.run_params['params']['eval_metric']
      )]
    )`

Needs list comprehension, also data_name is a (Dmatrix, string) pair so you'll need data_name[1], and run.values.keys() is a superset because it also records 'Training time' (and possibly more stuff in future)

Reply:
good catch.